torchgeo/models/farseg.py

```diff
@@ -71,7 +73,20 @@ def __init__(
             max_channels = 2048
         else:
             raise ValueError(f"unknown backbone: {backbone}.")
-        self.backbone = getattr(resnet, backbone)(pretrained=backbone_pretrained)
+        kwargs = {}
+        if parse(torchvision.__version__) >= parse("0.12"):
+            if backbone_pretrained:
+                kwargs = {
+                    "weights": getattr(
+                        torchvision.models, f"ResNet{backbone[6:]}_Weights"
+                    ).DEFAULT
+                }
+            else:
+                kwargs = {"weights": None}
+        else:
+            kwargs = {"pretrained": backbone_pretrained}
```

Comment:
Much uglier than I would like but don't know of a cleaner way to support both old and new torchvision